bit9_connector.py

```diff
@@ -462,11 +522,86 @@ def _upload_file(self, param):
             summary = action_result.update_summary({'upload_status': upload_status})
             try:
                 summary['upload_status_desc'] = self.UPLOAD_STATUS_DESCS[str(upload_status)]
-            except:
+            except Exception:
                 pass
 
         return action_result.set_status(phantom.APP_SUCCESS)
 
+    def _list_files(self, param):
+        action_result = self.add_action_result(ActionResult(param))
+        ret_val, limit = self._validate_integer(action_result, param.get('limit', 100), 'Limit', False)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        params = {
+            'limit': limit
+        }
+
+        endpoint = FILE_UPLOAD_ENDPOINT
+
+        ret_val, resp_json = self._make_rest_call(endpoint, action_result, method="get", params=params)
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        action_result.add_data(resp_json)
+        if limit == -1:
+            total = resp_json['count']
+            action_result.update_summary({'total': total})
+            return action_result.set_status(phantom.APP_SUCCESS, "Total: {}".format(total))
```

Comment:
Should the field `num_files` should be included in both conditions?

Reply:
Hey @paulnawat, The API has functionality that for limit = -1, it will return only the count of the files that are available on the server. In this case, only the total number will be returned without any other data. Therefore we have added different summery in both the cases.